### PR TITLE
3.4/n example overhaul: dbt_python_assets to recommended project layout + rename

### PR DESCRIPTION
**Pull request by yuhan**
### Summary & Motivation
- renamed it to starting with `assets_*` (plan to rename all assets related to `assets_*` so they could show up nicely in the examples folder - in alphabetical order): dbt_python_assets -> assets_dbt_python
- converted the new folder structure (scaffolded this example with `dagster project scaffold` and ported the contents over)
- updated [README](https://github.com/dagster-io/dagster/tree/yuhan/07-31-example_overhaul_dbt_python_assets_to_recommended_project_layout/examples/assets_dbt_python) to include desc and dagster project from-example cmd

see full overhaul plan in [this spreadsheet](https://elementl.quip.com/bPbFArMkloDp/Dagster-Example-Overhaul#temp:C:CZV1c9ece717ec2477aaf338ba54)
and more diffs coming in stack later

### How I Tested These Changes
- tox
- `dagster project from-example --name my-dagster-project --example assets_dbt_python `
- dagit loads:
Screen Shot 2022-07-31 at 12 51 27 AM

### CODE REVIEW

**Comment by yuhan** [comment]
Current dependencies on/for this PR:
* master
  * **PR #9071** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9071" target="_blank">Graphite</a> 
    * **PR #9116** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9116" target="_blank">Graphite</a> 
      * **PR #9117** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9117" target="_blank">Graphite</a> 
        * **PR #9120** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9120" target="_blank">Graphite</a> 
          * **PR #9121** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9121" target="_blank">Graphite</a> 
            * **PR #9122** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9122" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/9122?utm_source=stack-comment).

**Review by OwenKephart — APPROVED** [review]
looks great :)

File: examples/assets_dbt_python/assets_dbt_python/repository.py [hunk 1]
```diff
@@ -1,9 +1,8 @@
 import os
 
-import dbt_python_assets.forecasting as forecasting
-import dbt_python_assets.raw_data as raw_data
+from assets_dbt_python.assets import forecasting, raw_data
```
Comment (jamiedemaria) [inline examples/assets_dbt_python/assets_dbt_python/repository.py:3]: I'm having a hard time keeping track of the moving pieces in this diff, but I can't seem to find any assets defined in the forecasting package

Comment (yuhan) [reply examples/assets_dbt_python/assets_dbt_python/repository.py:3]: it's in the __init__.py - btw using the github branch folder structure view might be the easiest way to review: https://github.com/dagster-io/dagster/blob/yuhan/07-31-example_overhaul_dbt_python_assets_to_recommended_project_layout/examples/assets_dbt_python/assets_dbt_python/assets/forecasting/__init__.py

the reason is we are doing load from modules, so forecasting and raw_data need to a "module" and i dont feel it's necessary to include another file in the module folder. happy to change if that's super confusing.

Comment (jamiedemaria) [reply examples/assets_dbt_python/assets_dbt_python/repository.py:3]: ok cool! github shows a completely empty __init__.py for me, which was throwing me off

**Pull request merged**